lib/ProkkaAnnotation/ProkkaAnnotationImpl.py

```diff
@@ -283,66 +310,367 @@ def annotate_contigs(self, ctx, params):
+                            for sso_item in sso_list:
+                                sso_terms[sso_item["id"]] = {"id": sso_item["id"],
+                                                             "evidence": [],
+                                                             "term_name": sso_item["name"],
+                                                             "ontology_ref": self.sso_ref,
+                                                             "term_lineage": []}
+
+                        gene_features["ontology_terms"] = sso_terms
+                genome[gid] = gene_features
+
+        return genome
+
+    def write_genome_to_fasta(self, genome_data):
+        """
+
+        :param genome_data:
+        :return:
+        """
+        fasta_for_prokka_filepath = os.path.join(self.scratch,
+                                                 "features_" + str(uuid.uuid4()) + ".fasta")
+        with open(fasta_for_prokka_filepath, "w") as f:
+            for item in genome_data["data"]["features"]:
+                if "id" not in item and "dna_sequence" not in item:
+                    print("COMPLAINING ABOUT THIS BAD SEQUENCE")
+                else:
+                    f.write(">" + item["id"] + "\n" + item["dna_sequence"] + "\n")
+        print("Finished printing to" + fasta_for_prokka_filepath)
+        return fasta_for_prokka_filepath
+
+    def annotate_genome_with_new_annotations(self, **annotation_args):
+        """
+
+        :param annotation_args: genome_data, new_annotations from prokka, and the output_genome_name
+        :type
+        :return:
+        """
+        genome_data = annotation_args["genome_data"]
+        new_annotations = annotation_args["new_annotations"]
+        output_genome_name = annotation_args["output_genome_name"]
+
+        stats = {"current_functions": len(genome_data["data"]["features"]), "new_functions": 0,
+                 "new_ontologies": 0}
+
+        function_report_filepath = os.path.join(self.scratch, "ontology_report")
+        ontology_report_filepath = os.path.join(self.scratch, "function_report")
+
+        onto_r = open(function_report_filepath, "w")
+        func_r = open(ontology_report_filepath, "w")
+
+        func_r.write("function_id current_function new_function\n")
+        onto_r.write("function_id current_ontology new_ontology\n")
+
+        for i, feature in enumerate(genome_data["data"]["features"]):
+            fid = feature["id"]
+            current_function = feature.get("function", "")
+            current_ontology = feature.get("ontology_terms", None)
+            new_function = ""
+            new_ontology = dict()
+            if fid in new_annotations:
+                new_function = new_annotations[fid].get("function", "")
+                new_ontology = new_annotations[fid].get("ontology_terms", None)
+
+                if new_function is not "" and new_function is not "hypothetical protein":
```

Comment:
empty strings are falsey so can just say `if new _function` though I suppose this is more explicit

Reply:
fixed this, expanded scope of string checking too